File: 1165.Single-Row Keyboard.java
class Solution {
    public int calculateTime(String keyboard, String word) {
        Map<Integer, Integer> map = new HashMap<>();
        keyboard.chars().reduce(0, (index, c) -> {map.put(c, index); return index + 1;});
        List<Integer> lastCharWrapper = Arrays.asList(new Integer[]{0});
        return word.chars().reduce(
            0,
            (acc, c) -> {
                acc += Math.abs(lastCharWrapper.get(0) - map.get(c));
                lastCharWrapper.set(0, map.get(c));
                return acc;
            }
        );
    }
}

File: 1001-1500/1165.Single-Row Keyboard.py
class Solution(object):
    def calculateTime(self, keyboard, word):
        dic = dict(map(reversed, enumerate(keyboard)))
        posi = map(dic.__getitem__, word)
        return sum(map(
            lambda (a, b): abs(a - b),
            zip(posi, [0] + posi[:-1])
        ))
